bin/metrics-pdns.rb

```diff
@@ -37,15 +37,51 @@ class PdnsGraphite < Sensu::Plugin::Metric::CLI::Graphite
          long: '--scheme SCHEME',
          default: "#{Socket.gethostname}.pdns"
 
-  def run
-    metrics = `sudo rec_control get-all`.split("\n")
+  option :syslog_path,
+         description: 'Path to the syslog',
+         long: '--syslog SYSLOG',
+         default: '/var/log/messages'
+
+  option :extra_stats,
+         description: 'flag to send stats collected from syslog like qps',
+         short: '-e EXTRA',
+         long: '--extra EXTRA',
+         default: false
+
+  def cmd_run(cmd)
+    result = `#{cmd}`.split("\n")
     if $CHILD_STATUS.exitstatus > 0
-      critical 'Failed to dump pdns statistics. Check that the pdns process is running!'
+      critical "Failed to execute #{cmd}"
+    end
+    result
+  end
+
+  def parse_syslog
+    # send signal to pdns process to dump stats
+    cmd_run('sudo pkill -SIGUSR1 pdns_recursor')
+    # parse stats from syslog
+    metrics = cmd_run("sudo tail #{config[:syslog_path]} -n 6 | grep pdns_recursor")
+    return if metrics.nil?
+    metrics.each do |metric|
+      next unless metric.include?('stats:')
+      (_jargon, stats) = metric.split('stats:')
```

Comment:
I have to admit `_jargon` made me laugh.

Reply:
:)